### ISSUE TITLE
[question] Option to run asynchronous queries with the java API

### ISSUE THREAD

**Issue by q10** [arangodb/arangodb-java-driver#169]
In the HTTP API, there is the option to add "x-arango-async: store" to the headers so that a query can be run asynchronously, in the sense that a job id is returned and results can be fetched later.  Is this option also possible through the Java API?  I understand that the API already has Java futures, but I would not like my app to have many background threads waiting for results on queries.  Would it be possible to have the API return just a future of a job id, which the client can reference at a later time to fetch the results?

**Comment by mpv1989** [comment]
I understand your point of view. This sounds like a valid use case but we don't have a good implementation story for this kind of feature at this time.

**Comment by rocketraman** [comment]
Agreed, the async driver isn't really async. Its an async API on top of a synchronous driver.
